rest/server_context.go

```diff
@@ -40,12 +41,13 @@ const DefaultStatsLogFrequencySecs = 60
 // This struct is accessed from HTTP handlers running on multiple goroutines, so it needs to
 // be thread-safe.
 type ServerContext struct {
-	config       *ServerConfig
-	databases_   map[string]*db.DatabaseContext
-	lock         sync.RWMutex
-	statsContext *statsContext
-	HTTPClient   *http.Client
-	replicator   *base.Replicator
+	config           *ServerConfig
+	databases_       map[string]*db.DatabaseContext
+	lock             sync.RWMutex
+	statsContext     *statsContext
+	HTTPClient       *http.Client
+	replicator       *base.Replicator
+	pprofFileHandler *os.File
```

Comment:
If we're going with this approach, I think we need a mutex around it. It's possible for 2 HTTP requests to simultaneously read/write.

Reply:
This is too intrusive for a test-only fix.  Is this an actual functional issue when running an asynchronous cpu profile on Windows?

Reply:
Fixed in latest revision